sources/packages/backend/libs/test-utils/src/factories/application-offering-change-request.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  ApplicationOfferingChangeRequest,
+  ApplicationOfferingChangeRequestStatus,
+  ApplicationStatus,
+  EducationProgramOffering,
+  InstitutionLocation,
+  User,
+} from "@sims/sims-db";
+import { saveFakeApplicationDisbursements } from "./application";
+import { DataSource } from "typeorm";
+import { createFakeEducationProgramOffering } from "./education-program-offering";
+import { createFakeUser } from "./user";
+import * as faker from "faker";
+
+/**
+ * Create and save a fake application offering request change record.
+ * @param dataSource manages the repositories to save the data.
+ * @param relations dependencies:
+ * - `institutionLocation` related location.
+ * @param options additional options:
+ * - `applicationOfferingChangeRequestStatus` application offering change request status.
+ * @returns created and saved application offering request change record.
+ */
+export async function saveFakeApplicationOfferingRequestChange(
+  dataSource: DataSource,
+  relations?: {
+    institutionLocation?: InstitutionLocation;
+  },
+  options?: {
+    applicationOfferingChangeRequestStatus?: ApplicationOfferingChangeRequestStatus;
+  },
+): Promise<ApplicationOfferingChangeRequest> {
+  const userRepo = dataSource.getRepository(User);
+  const savedUser = await userRepo.save(createFakeUser());
+
+  const applicationOfferingChangeRequestRepo = dataSource.getRepository(
+    ApplicationOfferingChangeRequest,
+  );
+  const offeringRepo = dataSource.getRepository(EducationProgramOffering);
+  const requestedOffering = await offeringRepo.save(
+    createFakeEducationProgramOffering({
+      ...relations,
+      auditUser: savedUser,
+    }),
+  );
+  const application = await saveFakeApplicationDisbursements(
+    dataSource,
+    relations,
+    {
+      applicationStatus: ApplicationStatus.Completed,
+    },
+  );
+
+  const applicationOfferingChangeRequest =
+    new ApplicationOfferingChangeRequest();
+  applicationOfferingChangeRequest.application = application;
+  applicationOfferingChangeRequest.requestedOffering = requestedOffering;
+  applicationOfferingChangeRequest.activeOffering =
+    application.currentAssessment.offering;
+  applicationOfferingChangeRequest.applicationOfferingChangeRequestStatus =
+    options?.applicationOfferingChangeRequestStatus ??
+    ApplicationOfferingChangeRequestStatus.InProgressWithStudent;
+  applicationOfferingChangeRequest.reason = faker.lorem.sentence();
```

Comment:
Should we limit the amount of random text generated to ensure that will be less than 500?

Reply:
there is nothing for the character limit, I have added `word limit = 3). 
![image](https://github.com/bcgov/SIMS/assets/77353155/41b2db66-7e4c-4d2a-a578-29a9d103f593)

Reply:
That works also @ann-aot .